addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd

```diff
@@ -379,27 +383,111 @@ func is_angle_in_interval(angle: float, direction_angle: ESCDirectionAngle) -> b
 #
 # - deg int angle to set the character
 # - immediate
-#	If true, direction is switched immediately. Else, successive animations are
-#	used so that the character turns to target angle.
+#	If true, direction is switched immediately. Else, successive 
+#	animations are used so that the character turns to target angle.
 func set_angle(deg: int, immediate = true) -> void:
 	if deg < 0 or deg > 360:
 		escoria.logger.report_errors(
 			"movable.gd:set_angle()",
 			["Invalid degree to turn to " + str(deg)]
 		)
 	moved = true
-	last_deg = deg
-	last_dir = _get_dir_deg(deg, parent.animations)
-
-	# The character may have a state animation from before, which would be
-	# resumed, so we immediately force the correct idle animation
-	if parent.animation_sprite.animation != \
-			parent.animations.idles[last_dir].animation:
-		parent.animation_sprite.play(parent.animations.idles[last_dir].animation)
-	pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+	
+	if immediate:
+		last_deg = deg
+		last_dir = _get_dir_deg(deg, parent.animations)
+
+		# The character may have a state animation from before, which would be
+		# resumed, so we immediately force the correct idle animation
+		if parent.animation_sprite.animation != \
+				parent.animations.idles[last_dir].animation:
+			parent.animation_sprite.play(
+				parent.animations.idles[last_dir].animation
+			)
+		pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+	else:
+		var current_dir = last_dir
+		last_deg = deg
+		var target_dir = _get_dir_deg(deg, parent.animations)
+		
+		if target_dir - current_dir > 1:
```

Comment:
What if the target_dir is smaller than current_dir? Then this would be < 1...

Or am I not getting something here?

Reply:
Indeed, this test has no use. Same for the else block after it. I'll fix this.